File: 1894-merge-strings-alternately/1894-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder str = new StringBuilder();
        String fullWord = word1 + word2;
        int len1 = word1.length(), len2 = word2.length(), i = 0;
        int n = Math.max(len1, len2);
        
        while(i < n){
            if(i<len1 && i<len2){
                str.append(word1.charAt(i));
                str.append(word2.charAt(i));
            } else if(i>=len1 && i<len2){
                str.append(word2.charAt(i));
            } else if(i<len1 && i>=len2){
                str.append(word1.charAt(i));
            }
            i++;
        }
        
        return str.toString();
    }
}

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let mergedWord = "";
    let len1 = word1.length, len2 = word2.length, i = 0;
    while(i<len1 || i<len2){
        if(i<len1)
            mergedWord+=word1[i];
        if(i<len2)
            mergedWord+=word2[i];
        i++;
    }

    return mergedWord;
};